addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1942,7 +1962,14 @@ var SnippetsMenu = Widget.extend({
                     'drop-near': option['drop-near'] ? option['drop-near'].all().length : 0,
                     'drop-in': option['drop-in'] ? option['drop-in'].all().length : 0
                 };
-                check = (cache[k]['drop-near'] || cache[k]['drop-in']);
+                dropNear = cache[k]['drop-near'];
+                check = (dropNear || cache[k]['drop-in']);
+                if (check && !dropNear && excludeDropInOptions.length) {
+                    const excludeDropInSelector = excludeDropInOptions.map(el => { 
+                        return el["exclude-drop-in"];
+                    }).join(", ");
+                    check = option['drop-in'].all().not(excludeDropInSelector).length;
+                }
```

Comment:
Oh ok I think I understand now why you did not modify the `all` etc method directly... you allow to add an exclude on another definition as the "drop-in" definition... it make sense but it really complicates things for such a small bug :grimacing: 

-> Here you had to duplicate the logic of the drag and drop as you have to do the job of the drag & drop searching for all options that matches the base_selector, getting their exclude-drop-in, etc (the drag and drop logic is already badly duplicated for some of this logic, let's not de-deduplicate things even more)